File: timeConversion.js
/*
convert AM/PM time to 
military time
*/

let time = '06:40:03 AM'


function timeConversion(time) {
    let timeList = time.split(':')
    let hour = parseInt(timeList[0])
    if (hour >= 1 && hour <= 11 && time.includes("P")) {
        let milHour = hour + 12
        let milTime = milHour
        return (`${milTime}${time.substr(2, 6)}`)
    }
    else if (hour === 12 && time.includes("A")) {
        let midnight = '00'
        let milTime = midnight
        return (`${milTime}${time.substr(2, 6)}`)
    }
    else {
        return (time.substr(0, 8))
    }
}

console.log(timeConversion(time))

File: time_conversion.py
# TIME CONVERSION #


# convert from AM/PM to mil time

time = '06:40:03 PM'


def time_conversion(time):
    time_list = list(time[0:2])
    time_list = [''.join(time_list)]
    hour = [int(i) for i in time_list]
    if 1 <= hour[0] <= 11 and 'P' in time:
        mil_hour = hour[0] + 12
        mil_time = str(mil_hour) + time[2:8]
        return mil_time
    elif hour[0] == 12 and 'A' in time:
        hour[0] = '00'
        midnight = str(*hour) + time[2:8]
        return midnight
    else:
        return time[:8]


print(time_conversion('07:19:32 PM'))
